src/session.rs

```diff
@@ -337,7 +343,16 @@ where
                         );
                     }
 
-                    Ok(None) => {}
+                    // ...otherwise `store_session` returns `None`.
+                    Ok(None) => {
```

Comment:
@Ptrskay3 I think this is what was missing from our prior patch re this particular case. If you have a chance, would love your feedback on if this makes sense.